server/src/main/java/org/elasticsearch/action/admin/indices/settings/get/GetSettingsResponse.java

```diff
@@ -71,5 +111,135 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeString(cursor.key);
             Settings.writeSettingsToStream(cursor.value, out);
         }
+        if (out.getVersion().onOrAfter(org.elasticsearch.Version.V_7_0_0_alpha1)) {
+            out.writeVInt(indexToDefaultSettings.size());
+            for (ObjectObjectCursor<String, Settings> cursor : indexToDefaultSettings) {
+                out.writeString(cursor.key);
+                Settings.writeSettingsToStream(cursor.value, out);
+            }
+        }
+    }
+
+    private static void validateSettingsFieldName(XContentParser parser) throws IOException {
+        String settingsFieldName = parser.currentName();
+        XContentParserUtils.ensureExpectedToken(org.elasticsearch.common.xcontent.XContentParser.Token.FIELD_NAME, parser.currentToken(),
+            parser::getTokenLocation);
+        if (settingsFieldName.equals("settings") == false && settingsFieldName.equals("defaults") == false) {
+            String message = "Failed to parse object: expecting field with name [%s] but found [%s]";
+            throw new org.elasticsearch.common.ParsingException(parser.getTokenLocation(), String.format(java.util.Locale.ROOT, message,
```

Comment:
this makes parsing strict, which is good anywhere we do parsing in Elasticsearch besides response parsing :) that's because we would like the high-level REST client to work also against future versions of Elasticsearch, which may be adding new fields/arrays/objects to their responses.

Reply:
makes good sense